src/gap_filler.py

```diff
@@ -1,43 +1,103 @@
+import math as maths
 import random
 import spacy
-import utils
+from src import utils
+
```

Comment:
We already talked about user defined exceptions but the syntax you want is like this:

`class GapFinderException(Exception):`
    `    def __init__(self, message):`
        `        self.message= message`
then to use them:

`raise GapFinderException('whatever message will be useful`

you can also define more specific ones which make them easier to deal with like :

`class TextTooSmall(GapFinderException)`